app/src/main/java/org/xbmc/kore/service/library/SyncMusic.java

```diff
@@ -347,12 +348,28 @@ public void onSuccess(ApiList<AudioType.DetailsSong> result) {
 
                 // Save partial results to DB
                 ContentValues songValuesBatch[] = new ContentValues[items.size()];
+                int artistsCount = 0;
                 for (int i = 0; i < items.size(); i++) {
                     AudioType.DetailsSong song = items.get(i);
                     songValuesBatch[i] = SyncUtils.contentValuesFromSong(hostId, song);
+
+                    artistsCount += song.artistid.size();
                 }
                 contentResolver.bulkInsert(MediaContract.Songs.CONTENT_URI, songValuesBatch);
 
+                // Iterate on each song, collect the artists and insert them
+                ContentValues songArtistsValuesBatch[] = new ContentValues[artistsCount];
+                int artistCount = 0;
+                for (AudioType.DetailsSong song : items) {
+                    for (int artistId : song.artistid) {
+                        songArtistsValuesBatch[artistCount] = new ContentValues();
+                        songArtistsValuesBatch[artistCount].put(MediaContract.SongArtists.HOST_ID, hostId);
+                        songArtistsValuesBatch[artistCount].put(MediaContract.SongArtists.SONGID, song.songid);
+                        songArtistsValuesBatch[artistCount].put(MediaContract.SongArtists.ARTISTID, artistId);
+                        artistCount++;
+                    }
+                }
+
```

Comment:
1. I suggest changing the `artistsCount`  (plural) variable to `totalArtistsCount` to make it distinct from `artistCount`. It was confusing me.

2. You're populating `songArtistsValuesBatch`, but you're not doing anything with it! The songs aren't getting saved to the DB, and no songs are appearing on the UI. You might have missed this because you have a version with the `songsArtists` table filled, but you should test the common scenario, which is someone who is upgrading from a DB without that table.